tools/istio-iptables/pkg/e2e/e2e_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+	"testing"
+
+	// Create a new mount namespace.
+	"github.com/howardjohn/unshare-go/mountns"
+	// Create a new network namespace. This will have the 'lo' interface ready but nothing else.
+	_ "github.com/howardjohn/unshare-go/netns"
+	"github.com/howardjohn/unshare-go/userns"
+
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/pkg/slices"
+	"istio.io/istio/pkg/test/util/assert"
+	cleancmd "istio.io/istio/tools/istio-clean-iptables/pkg/cmd"
+	iptablescmd "istio.io/istio/tools/istio-iptables/pkg/cmd"
+)
+
+func TestIptablesCleanRoundTrip(t *testing.T) {
+	setup(t)
+	roundtrip := func(t *testing.T, args ...string) {
+		// We only have UID 0 in our namespace, so always set it to that
+		args = append([]string{"--proxy-uid=0"}, args...)
+		// Apply, cleanup with the old approach
+		assert.NoError(t, runIptables(args...))
+		assert.NoError(t, runIptablesOldClean(args...))
+		validateIptablesClean(t)
+		// App, cleanup with the new approach
+		assert.NoError(t, runIptables(args...))
+		assert.NoError(t, runIptablesClean(args...))
```

Comment:
The --cleanup-only option is already verified in TestIdempotentUnequaledRerun in the file istio-iptables/pkg/capture/run_test.go.
https://github.com/istio/istio/blob/master/tools/istio-iptables/pkg/capture/run_test.go#L384-L393

The difference between the two is that the one in run_test.go uses the cleanup option programmatically by setting the istio-iptables/config instead of relying on the CLI.
Do you prefer to have a verification via CLI as well?

Reply:
hmm I guess we probably don't, I missed that part of your PR

Reply:
This does test `istio-clean-iptables` as well though

Reply:
yes that's true. The old istio-clean-iptables method has currently zero e2e coverage